pkg/storegateway/bucket_stores.go

```diff
@@ -547,8 +549,10 @@ func (u *BucketStores) getOrCreateStore(userID string) (*store.BucketStore, erro
 	userBkt := bucket.NewUserBucketClient(userID, u.bucket, u.limits)
 	fetcherReg := prometheus.NewRegistry()
 
+	ignoreBlocksBefore := -model.Duration(u.cfg.BucketStore.IgnoreBlocksBefore)
 	// The sharding strategy filter MUST be before the ones we create here (order matters).
 	filters := append([]block.MetadataFilter{NewShardingMetadataFilterAdapter(userID, u.shardingStrategy)}, []block.MetadataFilter{
+		block.NewTimePartitionMetaFilter(thanos_model.TimeOrDurationValue{Dur: &ignoreBlocksBefore}, thanos_model.TimeOrDurationValue{}),
```

Comment:
I dont see in the source code a handling for 0 to disable it. Am i missing it?

It seems for max we are always returning 0
https://github.com/thanos-io/thanos/blob/main/pkg/model/timeduration.go#L73

And all blocks will be filter at 
https://github.com/thanos-io/thanos/blob/main/pkg/block/fetcher.go#L690